build_daemon/lib/change_provider.dart

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:watcher/src/watch_event.dart';
+
+abstract class ChangeProvider {}
```

Comment:
I'd be more comfortable if both methods were defined here and we have a doc comment that says it's fine for one method to always return "empty" if the other method encompasses all changes.

In the long term we can also consider going through the `collectChanges` path even if the build was triggered by something on the `changes` stream with the intent of hypothetical use cases like "we auto rebuild if you edit in `lib/`, but if you edit somewhere else you need to manually trigger a build"

Reply:
Done.